### PR TITLE
:es6: live blog updates

### PR DESCRIPTION
**Pull request by philwills**
## What does this change?
✨ ES6 ✨ the live blog update module for fronts. 

This follows up the bug fix in #17271 

## What is the value of this and can you measure success?
✨ 

## Does this affect other platforms - Amp, Apps, etc?
No

## Tested in CODE?
No, locally.

### CODE REVIEW

File: static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+const viewportHeightPx = detect.getViewport().height;
+
+type Block = {
+    id: string,
+    title: string,
+    publishedDateTime: number,
+    lastUpdatedDateTime: number,
+    body: string,
+    isNew: boolean, // This is not pulled in from the response, but mutated in this module
+};
```
Comment (SiAdcock) [inline static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js]: 🎉

File: static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            block.id &&
+            block.publishedDateTime &&
+            block.body &&
+            block.body.length >= 10
+    );
+
+const showUpdatesFromLiveBlog = (): Promise<void> =>
+    fastdomPromise
+        .read(() => {
+            const elementsById: Map<string, Array<Element>> = new Map();
```
Comment (SiAdcock) [inline static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js]: What do we gain by using a `Map` here instead of an object literal?

**Review by SiAdcock — APPROVED** [review]
Looks great!

File: static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js (hunk 2 above)
Comment (philwills) [inline static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js]: - `forEach` on a `Map` iterates over the key and value, as opposed to an Object literal where you can get the keys. This allowed for a more direct translation from the underscore equivalent.
- Defining it as a `Map` also allows you to be explicit that this homogenous structure where all the keys and all the values are of the same type. 
- I'd use an object literal where the keys were statically known, but it seems more natural to use a `Map` where the keys are dynamically assigned.

**Pull request merged**